presto-cli/src/main/java/com/facebook/presto/cli/ErrorMessages.java

```diff
@@ -14,22 +14,55 @@
 package com.facebook.presto.cli;
 
 import com.facebook.presto.client.ClientSession;
+import com.facebook.presto.client.ErrorLocation;
+import com.facebook.presto.client.QueryError;
+import com.facebook.presto.client.QueryResults;
+import com.facebook.presto.client.StatementClient;
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import org.fusesource.jansi.Ansi;
 
 import java.io.PrintWriter;
 import java.io.StringWriter;
+import java.util.List;
+
+import static com.facebook.presto.cli.ConsolePrinter.REAL_TERMINAL;
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.String.format;
 
 public class ErrorMessages
 {
+    private static final String TECHNICAL_DETAILS_HEADER = "\n=========   TECHNICAL DETAILS   =========\n";
+    private static final String SESSION_INTRO = "[ Session information ]\n";
+    private static final String ERROR_MESSAGE_INTRO = "[ Error message ]\n";
+    private static final String STACKTRACE_INTRO = "[ Stack trace ]\n";
+    private static final String TECHNICAL_DETAILS_END = "========= TECHNICAL DETAILS END =========\n\n";
+
     private ErrorMessages()
     {}
 
-    public static String createErrorMessage(Throwable throwable, ClientSession session)
+    public static String createQueryErrorMessage(StatementClient client)
```

Comment:
It's not only about error. This class is very specific and depends on many other classes anyway. It is meant to recognise multiple specific exception classes that only make sense in the context of Presto.

This particular dependency allow to provide better error messages by extracting informations like query, debug mode, and more. Generalising that class is a cool project for an external library, I didn't like the idea of over engineering simple things we want to do in this class though.

If you prefer generalised way of doing that that will (for example) allow to map particular parameters from outside of class and make the process of error recognition dynamic, please let me know - I can do that with pleasure - I just didn't thought it's worth it :)

I can also to something in between and only get rid of StatementClient dependency, but my opinion is still to leave it as is.

Please let me know what you decide.